src/main/scala/scorex/core/network/RequestTracker.scala

```diff
@@ -0,0 +1,102 @@
+package scorex.core.network
+
+import akka.actor._
+import akka.util.Timeout
+import scorex.core.network.NetworkController.ReceivableMessages.{GetConnectedPeers, PenalizePeer, RegisterMessageSpecs, SendToNetwork}
+import scorex.core.network.message.Message
+import scorex.core.network.message.Message.MessageCode
+import scorex.core.network.peer.PenaltyType
+import scorex.core.utils.ActorHelper
+import scorex.util.ScorexLogging
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.{FiniteDuration, SECONDS}
+import scala.language.postfixOps
+
+/**
+  * Wraps NetworkController actor, providing verification of request-response communication with other peers
+  */
+class RequestTracker(
+                      networkControllerRef: ActorRef,
+                      trackedRequestCode: MessageCode,
+                      trackedResponseCode: MessageCode,
+                      deliveryTimeout: FiniteDuration,
+                      penalizeNonDelivery: Boolean)(implicit ec: ExecutionContext)
+  extends Actor with ScorexLogging with ActorHelper {
+
+  private var handler: Option[ActorRef] = None
+
+  private val requestTracker = mutable.Set.empty[(MessageCode, ConnectedPeer)]
+
+  implicit val timeout: Timeout = Timeout(60, SECONDS)
+
+  override def receive: Receive = {
+    registerMessageSpec orElse
+      sendTrackedRequest orElse
+      receiveResponse orElse
+      verifyDelivery orElse
+      forward
+  }
+
+  private def registerMessageSpec: Receive = {
+    case RegisterMessageSpecs(specs, handler) =>
+      this.handler = Some(handler)
+      networkControllerRef ! RegisterMessageSpecs(specs, self)
+  }
+
+  private def sendTrackedRequest: Receive = {
+    case message@SendToNetwork(m@Message(spec, _, _), strategy) if spec.messageCode == trackedRequestCode =>
+      askActor[Seq[ConnectedPeer]](networkControllerRef, GetConnectedPeers)
+        .map { peers =>
+          strategy.choose(peers.filter(_.peerInfo.exists(_.peerSpec.protocolVersion >= m.spec.protocolVersion)))
+            .foreach { peer =>
+              val requestKey = (trackedRequestCode, peer)
+
+              requestTracker += requestKey
```

Comment:
For that to be possible, we have to be able to distinguish request-response pairs from one another. 
This is because when checking delivery after 60 seconds, we have to know which one of 2 requests were answered. 
So, we have to choose between:
1. Supporting several requests and not having delivery checks == possible memory leak
2. Keeping delivery check and only tracking one request at a time.

I don't know a good way to overcome it now, I hope the idea will come with time spent on scorex and better understanding the system.